### PR TITLE
Port the water example shaders to WGSL

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

**Comment by kvark** [comment]
Currently blocked on:
  - proper `modf` implementation, see https://github.com/gpuweb/gpuweb/issues/1480
  - `Matrix` SPIR-V capability in SPIR-V frontend

File: examples/README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ All framework-based examples render to the window.
 ## Feature matrix
 | Feature                      | boids  | cube   | mipmap | msaa-line | shadow | skybox | texture-arrays | water  | conservative-raster |
 | ---------------------------- | ------ | ------ | ------ | --------- | ------ | ------ | -------------- | ------ | ------------------- |
-| WGSL shaders                 | :star: | :star: | :star: | :star:    | :star: | :star: |                |        | :star:              |
+| WGSL shaders                 | :star: | :star: | :star: | :star:    | :star: | :star: |                | :star: | :star:              |
```
Comment (cwfitzgerald) [inline examples/README.md:15]: So close!!!

**Review by cwfitzgerald — APPROVED** [review]
Man it's so fun to see both shaders in one file!

bors r+

**Comment by kvark** [comment]
bors r=cwfitzgerald

File: examples/README.md (hunk 1 above)
Comment (cwfitzgerald) [reply examples/README.md:15]: Actually yeah that makes a lot of sense!

**Pull request merged**